From e035a9b42d0f07ba1e153f89891858b7ba2e7609 Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:19 +0100
Subject: [PATCH 26/51] Revert "survive -Inon-accessible-directory:"

This reverts commit 02f1b4b8cb15ddb132f2ccb2ab3109523e19fb74.
---
 interpreter/llvm-project/clang/lib/Lex/HeaderSearch.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/interpreter/llvm-project/clang/lib/Lex/HeaderSearch.cpp b/interpreter/llvm-project/clang/lib/Lex/HeaderSearch.cpp
index 7ab8527619..dfa974e9a6 100644
--- a/interpreter/llvm-project/clang/lib/Lex/HeaderSearch.cpp
+++ b/interpreter/llvm-project/clang/lib/Lex/HeaderSearch.cpp
@@ -426,9 +426,7 @@ OptionalFileEntryRef HeaderSearch::getFileAndSuggestModule(
     std::error_code EC = llvm::errorToErrorCode(File.takeError());
     if (EC != llvm::errc::no_such_file_or_directory &&
         EC != llvm::errc::invalid_argument &&
-        EC != llvm::errc::is_a_directory &&
-        EC != llvm::errc::not_a_directory &&
-        EC != llvm::errc::permission_denied) {
+        EC != llvm::errc::is_a_directory && EC != llvm::errc::not_a_directory) {
       Diags.Report(IncludeLoc, diag::err_cannot_open_file)
           << FileName << EC.message();
     }
-- 
2.43.5

